internal/entity/migrate/dbms_authid_migration.go

```diff
@@ -0,0 +1,169 @@
+				return fmt.Errorf("migrate: error getting index list %w", err)
+			}
+			for _, index := range indexes {
+				dropStatement := fmt.Sprintf(`DROP INDEX IF EXISTS "%s"`, index.Name)
+				if err := db.Exec(dropStatement).Error; err != nil {
+					return fmt.Errorf("migrate: error dropping index %s was %w", index.Name, err)
+				}
+			}
+		}
+		if !db.HasTable("auth_users") {
+			if err := db.Exec(authUsersCreate).Error; err != nil {
+				return fmt.Errorf("migrate: error creating auth_users %w", err)
+			}
+		} else {
+			// Data Migration here, by rename, create new, data transfer, drop indexes
+			if err := db.Exec(`ALTER TABLE "auth_users" RENAME TO "migrate_auth_users"`).Error; err != nil {
+				return fmt.Errorf("migrate: error renaming auth_users %w", err)
+			}
+			if err := db.Exec(authUsersCreate).Error; err != nil {
+				return fmt.Errorf("migrate: error creating auth_users %w", err)
+			}
+
+			// Get the columns of both old and new table, and find the columns that are in the old and new table
+			var oldPragmaColumns []pragmaTable
+			var newPragmaColumns []pragmaTable
+			oldColumns := make(map[string]bool)
+
+			if err := db.Raw("PRAGMA table_info(migrate_auth_users)").Scan(&oldPragmaColumns).Error; err != nil {
+				return fmt.Errorf("migrate: error getting column list for migrate_auth_users with %w", err)
+			}
+			for _, pragma := range oldPragmaColumns {
+				oldColumns[pragma.Name] = false
+			}
+
+			if err := db.Raw("PRAGMA table_info(auth_users)").Scan(&newPragmaColumns).Error; err != nil {
+				return fmt.Errorf("migrate: error getting column list for auth_users with %w", err)
+			}
+			for _, pragma := range newPragmaColumns {
+				if _, present := oldColumns[pragma.Name]; present {
+					oldColumns[pragma.Name] = true
+				}
+			}
+			// Build the select into statement
+			var columns []string
+			for key, value := range oldColumns {
+				if value {
+					columns = append(columns, key)
+				}
+			}
+
+			populateStmt := fmt.Sprintf("INSERT INTO auth_users (%s) SELECT %s FROM migrate_auth_users", strings.Join(columns, ", "), strings.Join(columns, ", "))
+
+			if err := db.Exec(populateStmt).Error; err != nil {
+				return fmt.Errorf("migrate: error migrating with stmt %s with %w", populateStmt, err)
+			}
+
+			var indexes []resultIndex
+			if err := db.Raw("SELECT name FROM sqlite_master WHERE type = 'index' AND tbl_name = ? AND sql IS NOT NULL", "migrate_auth_users").Scan(&indexes).Error; err != nil {
+				return fmt.Errorf("migrate: error getting index list %w", err)
+			}
+			for _, index := range indexes {
+				dropStatement := fmt.Sprintf(`DROP INDEX IF EXISTS "%s"`, index.Name)
+				if err := db.Exec(dropStatement).Error; err != nil {
+					return fmt.Errorf("migrate: error dropping index %s was %w", index.Name, err)
+				}
+			}
+		}
+	case MySQL: // MySQL
+		// These create statements will get out of date, but that is ok, as the main migrate path will add any missing columns/indexes in later.
```

Comment:
What is the reasoning behind the MariaDB/MySQL migrations? How do the tables/schema created automatically by GORM differ? 🤔